packages/farrow-schema/src/validator.ts

```diff
@@ -405,6 +405,13 @@ Validator.impl(S.Unknown, {
   validate: (input) => Ok(input),
 })
 
+Validator.impl(S.Never, {
+  validate: (input) => {
+    if (input === undefined) return Ok(undefined as never)
```

Comment:
`never` is the empty type, not the unit type. it has zero value, so `undefined` is not a valid program for `never`

`throw new Error('There is no value allowed for Never')`, empty type means to terminate the program.

Reply:
Got  it.